vllm/worker/hpu_model_runner.py

```diff
@@ -956,45 +960,62 @@ def _prepare_prompt(
 
             prefix_block_list_tensor = torch.tensor(prefix_block_list,
                                                     dtype=torch.long,
-                                                    device=self.device)
+                                                    device='cpu',
+                                                    pin_memory=self.pin_memory)
         else:
             prefix_block_list_tensor = None
 
         input_tokens = make_tensor_with_pad(input_tokens,
                                             max_len=max_prompt_len,
                                             pad=0,
                                             dtype=torch.long,
-                                            device=self.device)
+                                            device='cpu',
+                                            pin_memory=self.pin_memory)
 
         input_positions = make_tensor_with_pad(input_positions,
                                                max_len=max_prompt_len,
                                                pad=0,
                                                dtype=torch.long,
-                                               device=self.device)
+                                               device='cpu',
+                                               pin_memory=self.pin_memory)
 
         slot_mapping = make_tensor_with_pad(slot_mapping,
                                             max_len=max_prompt_len,
                                             pad=_PAD_SLOT_ID,
                                             dtype=torch.long,
-                                            device=self.device)
+                                            device='cpu',
+                                            pin_memory=self.pin_memory)
 
         seq_lens_tensor = torch.tensor(seq_lens,
                                        dtype=torch.long,
-                                       device=self.device)
+                                       device='cpu',
+                                       pin_memory=self.pin_memory)
```

Comment:
does pinning memory change anything on HPU?

Reply:
Removed, does not change anything